packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -0,0 +1,103 @@
+type Value<T, K extends string> = K extends keyof T ? T[K] : unknown;
```

Comment:
A thing I'll check on tomorrow (when hopefully finishing the v4 DT push!)—we should make sure this (and similar on the `@ember/application` PR) aligns with the updated types there. For reference, PR is [here][DT]. The types shouldn't change between now and landing it; it's all just config stuff to get the major version landed.

[DT]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58060